File: PegasusWebV1/PegasusWeb/PegasusWeb/Entities/Tarea.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace PegasusWeb.Entities
{
    public class Tarea
    {
        public int Id { get; set; }

        public string Titulo { get; set; }

        public string Descripcion { get; set; }

        public DateTime? Fecha_Entrega { get; set; }

        public bool? Entregado { get; set; }

        public double? Calificacion { get; set;}

        public Materia? Materia { get; set; }
        public int? Id_Materia { get; set; }

        public Usuario? Alumno { get; set; }
        public int? Id_Alumno { get; set; }
    }
}

File: pegasusapp/src/models/tarea.ts
import { Usuario } from './usuario';
import { Materia } from './materia';

export interface Tarea {
  id: number;
  titulo: string;
  descripcion: string;
  fecha_Entrega?: string | null;
  entregado?: boolean | null;
  calificacion?: number | null;

  materia?: Materia | null;
  id_Materia?: number | null;

  alumno?: Usuario | null;
  id_Alumno?: number | null;
}
